pkg/resources/flow.go

```diff
@@ -87,11 +86,32 @@ func (f *Flow) Create() error {
 		}
 	}
 	options := interfaces.DependencyGraphOptions{
-		FlowName: f.originalName,
-		Args:     args,
+		FlowName:         f.originalName,
+		Args:             args,
+		FlowInstanceName: f.generatedName,
 	}
+
+	if destroy {
+		// Delete one replica
+		options.ReplicaCount = -1
```

Comment:
I renamed the `destroy` parameter to `decreaseReplicaCount` (and `constructGraph` function to `buildDependencyGraph`)